x-pack/plugins/security_solution/public/common/components/navigation/solution_grouped_nav/solution_grouped_nav_panel.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment, useCallback } from 'react';
+import {
+  EuiDescriptionList,
+  EuiDescriptionListDescription,
+  EuiDescriptionListTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFocusTrap,
+  EuiOutsideClickDetector,
+  EuiPanel,
+  EuiPortal,
+  EuiTitle,
+  EuiWindowEvent,
+  keys,
+  useIsWithinBreakpoints,
+} from '@elastic/eui';
+import classNames from 'classnames';
+import { groupedNavPanelStyles } from './solution_grouped_nav_panel.styles';
+import { PortalNavItem } from './solution_grouped_nav_item';
+import { useShowTimeline } from '../../../utils/timeline/use_show_timeline';
+
+interface SolutionGroupedNavPanelProps {
+  onClose: () => void;
+  title: string;
+  items: PortalNavItem[];
+}
+
+const SolutionGroupedNavPanelComponent: React.FC<SolutionGroupedNavPanelProps> = ({
+  onClose,
+  title,
+  items,
+}) => {
+  const [isTimelineBottomBarVisible] = useShowTimeline();
+  const isLargerBreakpoint = useIsWithinBreakpoints(['l', 'xl']);
+
+  const styles = groupedNavPanelStyles(isTimelineBottomBarVisible && isLargerBreakpoint);
+  const panelClasses = classNames('kbnPageTemplateSolutionNavPanel', 'eui-yScroll');
```

Comment:
Please re-name these classes and components to be security specific, otherwise it will be confusing to other devs where this component actually lives.

Reply:
I tried to implement this component as generic as possible, so we can easily move it to shared components in case other solutions also want to use it. Do you think I should name everything with "Security"?